internal/fields/validate.go

```diff
@@ -214,6 +214,25 @@ func initializeAllowedCIDRsList() (cidrs []*net.IPNet) {
 	return cidrs
 }
 
+// IsDocumentation reports whether ip is a reserved address for documentation,
+// according to RFC 5737 (IPv4 Address Blocks Reserved for Documentation) and
+// RFC 3849 (IPv6 Address Prefix Reserved for Documentation).
+func IsDocumentation(ip net.IP) (bool) {
+	if ip4 := ip.To4(); ip4 != nil {
+		// Following RFC 5737, Section 3. Documentation Address Blocks which says:
+		//   The blocks 192.0.2.0/24 (TEST-NET-1), 198.51.100.0/24 (TEST-NET-2),
+		//   and 203.0.113.0/24 (TEST-NET-3) are provided for use in
+		//   documentation.
+		return ((ip4[0] == 192 && ip4[1] == 0 && ip4[2] == 2) ||
+			(ip4[0] == 198 && ip4[1] == 51 && ip4[2] == 100) ||
+			(ip4[0] == 203 && ip4[1] == 0 && ip4[2] == 113))
+	}
+  // Following RFC 3849, Section 2. Documentation IPv6 Address Prefix which
+  // says:
+  //   The prefix allocated for documentation purposes is 2001:DB8::/32
```

Comment:
Indentation looks wrong here.

Reply:
You can run `make check-static` for linting.

Reply:
Errr, `make check-static` definitely whinging, but not in any way that's interpretable...
```
colin.stubbs@MBP-13-M2 elastic-package % make check-static
go build -ldflags "-X github.com/elastic/elastic-package/internal/version.CommitHash=`git describe --always --long --dirty` -X github.com/elastic/elastic-package/internal/version.BuildTime=`date +%s` -X github.com/elastic/elastic-package/internal/version.Tag=`(git describe --exact-match --tags 2>/dev/null || echo '') | tr -d '\n'`" -o elastic-package
go run golang.org/x/tools/cmd/goimports -local github.com/elastic/elastic-package/ -w .
go run honnef.co/go/tools/cmd/staticcheck ./...
go run github.com/elastic/go-licenser -license Elastic
go mod tidy
cd tools/readme; go run main.go
2023/02/03 12:20:07 generating command doc for benchmark...
2023/02/03 12:20:07 generating command doc for build...
2023/02/03 12:20:07 generating command doc for changelog...
2023/02/03 12:20:07 generating command doc for check...
2023/02/03 12:20:07 generating command doc for clean...
2023/02/03 12:20:07 generating command doc for create...
2023/02/03 12:20:07 generating command doc for dump...
2023/02/03 12:20:07 generating command doc for export...
2023/02/03 12:20:07 generating command doc for format...
2023/02/03 12:20:07 generating command doc for install...
2023/02/03 12:20:07 generating command doc for lint...
2023/02/03 12:20:07 generating command doc for profiles...
2023/02/03 12:20:07 generating command doc for promote...
2023/02/03 12:20:07 generating command doc for publish...
2023/02/03 12:20:07 generating command doc for report...
2023/02/03 12:20:07 generating command doc for service...
2023/02/03 12:20:07 generating command doc for stack...
2023/02/03 12:20:07 generating command doc for status [package]...
2023/02/03 12:20:07 generating command doc for test...
2023/02/03 12:20:07 generating command doc for uninstall...
2023/02/03 12:20:07 generating command doc for version...
README.md successfully written
git update-index --really-refresh
internal/fields/validate.go: needs update
make: *** [check-git-clean] Error 1
colin.stubbs@MBP-13-M2 elastic-package % 
```

Reply:
`make build` / `make` works fine... I'm already using a locally build version of `elastic-agent`

`make lint` reports no errors.

`make check-static` is whinging but unhelpful... 

```
user@box elastic-package % make check-static
go build -ldflags "-X github.com/elastic/elastic-package/internal/version.CommitHash=`git describe --always --long --dirty` -X github.com/elastic/elastic-package/internal/version.BuildTime=`date +%s` -X github.com/elastic/elastic-package/internal/version.Tag=`(git describe --exact-match --tags 2>/dev/null || echo '') | tr -d '\n'`" -o elastic-package
go run golang.org/x/tools/cmd/goimports -local github.com/elastic/elastic-package/ -w .
go run honnef.co/go/tools/cmd/staticcheck ./...
go run github.com/elastic/go-licenser -license Elastic
go mod tidy
cd tools/readme; go run main.go
2023/02/03 12:23:54 generating command doc for benchmark...
2023/02/03 12:23:54 generating command doc for build...
2023/02/03 12:23:54 generating command doc for changelog...
2023/02/03 12:23:54 generating command doc for check...
2023/02/03 12:23:54 generating command doc for clean...
2023/02/03 12:23:54 generating command doc for create...
2023/02/03 12:23:54 generating command doc for dump...
2023/02/03 12:23:54 generating command doc for export...
2023/02/03 12:23:54 generating command doc for format...
2023/02/03 12:23:54 generating command doc for install...
2023/02/03 12:23:54 generating command doc for lint...
2023/02/03 12:23:54 generating command doc for profiles...
2023/02/03 12:23:54 generating command doc for promote...
2023/02/03 12:23:54 generating command doc for publish...
2023/02/03 12:23:54 generating command doc for report...
2023/02/03 12:23:54 generating command doc for service...
2023/02/03 12:23:54 generating command doc for stack...
2023/02/03 12:23:54 generating command doc for status [package]...
2023/02/03 12:23:54 generating command doc for test...
2023/02/03 12:23:54 generating command doc for uninstall...
2023/02/03 12:23:54 generating command doc for version...
README.md successfully written
git update-index --really-refresh
internal/fields/validate.go: needs update
make: *** [check-git-clean] Error 1
user@box elastic-package % make build
go build -ldflags "-X github.com/elastic/elastic-package/internal/version.CommitHash=`git describe --always --long --dirty` -X github.com/elastic/elastic-package/internal/version.BuildTime=`date +%s` -X github.com/elastic/elastic-package/internal/version.Tag=`(git describe --exact-match --tags 2>/dev/null || echo '') | tr -d '\n'`" -o elastic-package
user@box elastic-package % 
```

Reply:
> Indentation looks wrong here.

Ewww, tabs in use... I've replaced the spaces and pushed another commit.